atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -149,6 +156,55 @@ def get_buildroot(self, worker_metadatas):
 
         return buildroots
 
+    def get_logs(self):
+        """
+        Build list of log files
+
+        :return: list, of log files
+        """
+
+        logs = None
+        output = []
+
+        # Collect logs from server
+        try:
+            logs = self.osbs.get_orchestrator_build_logs(self.build_id)
+        except OsbsException as ex:
+            self.log.error("unable to get build logs: %r", ex)
+            return output
+        except TypeError:
+            # Older osbs-client has no get_orchestrator_build_logs
+            self.log.error("Your osbs client doesn't support this functionality")
+            return output
+
+        platform_logs = {}
+        for entry in logs:
+            platform_logs.setdefault(entry.platform, [])
+            platform_logs[entry.platform].append(entry.line)
+
+        for arch in platform_logs:
+            filename = arch
+            if arch is None:
+                filename = "orchestrator"
+            # Delete once closed
+            logfile = NamedTemporaryFile(prefix="%s-%s" % (self.build_id, filename),
+                                         suffix=".log", mode='wb')
+            for entry in platform_logs[arch]:
+                logfile.write(entry)
+            logfile.flush()
+            metadata = self.get_output_metadata(logfile.name, "%s.log" % filename)
+            output.append(Output(file=logfile, metadata=metadata))
+
+        docker_logs = NamedTemporaryFile(prefix="docker-%s" % self.build_id,
+                                         suffix=".log", mode='wb')
+        docker_logs.write("\n".join(self.workflow.build_result.logs).encode('utf-8'))
```

Comment:
Isn't this empty on orchestrator builds?

`koji_upload` currently handles this file. We should probably change it there so it reports a build.log for each platform.

Reply:
Ah. It could well be empty, you'd know better than I would. Some of this is just copy-n-paste as a base from other logging code, it's entirely possible I've brought over bits that aren't needed. I *think* the $arch.log files are supposed to be the per-arch build.log files, aren't they? I could be woefully mistaken though.